### PR TITLE
CSP-1372 Get All Providers endpoint

### PR DESCRIPTION
**Pull request by MarkFCain**

### CODE REVIEW

File: src/Shared/SFA.DAS.SharedOuterApi/InnerApi/Responses/RoatpV2/GetProviderSummaryResponse.cs [hunk 1]
```diff
@@ -1,63 +1,5 @@
-namespace SFA.DAS.SharedOuterApi.InnerApi.Responses.RoatpV2
+namespace SFA.DAS.SharedOuterApi.InnerApi.Responses.RoatpV2;
+
+public class GetProviderSummaryResponse : Provider
 {
-    public class GetProviderSummaryResponse
```
Comment (DevenKShah) [inline src/Shared/SFA.DAS.SharedOuterApi/InnerApi/Responses/RoatpV2/GetProviderSummaryResponse.cs:3]: Delete commented code

File: src/EmployerPR/SFA.DAS.EmployerPR/Domain/RoatpProvider/GetRoatpProvidersResponse.cs [hunk 2]
```diff
@@ -0,0 +1,5 @@
+﻿namespace SFA.DAS.EmployerPR.Domain.RoatpProvider;
+public class GetRoatpProvidersResponse
```
Comment (DevenKShah) [inline src/EmployerPR/SFA.DAS.EmployerPR/Domain/RoatpProvider/GetRoatpProvidersResponse.cs:2]: What is this class for?

File: src/EmployerPR/SFA.DAS.EmployerPR.Api/AppStart/AddServiceCollectionExtensions.cs [hunk 3]
```diff
@@ -44,7 +44,8 @@ public static IServiceCollection AddServiceRegistration(this IServiceCollection
         services.AddTransient<IAccountsApiClient<AccountsConfiguration>, AccountsApiClient>();
         services.AddTransient<IEmployerProfilesApiClient<EmployerProfilesApiConfiguration>, EmployerProfilesApiClient>();
         services.AddTransient<IProviderRelationshipsApiClient<ProviderRelationshipsApiConfiguration>, ProviderRelationshipsApiClient>();
-
+        services.AddTransient<IRoatpV2TrainingProviderService, RoatpV2TrainingProviderService>();
+        services.AddTransient<IRoatpCourseManagementApiClient<RoatpV2ApiConfiguration>, RoatpCourseManagementApiClient>();
```
Comment (DevenKShah) [inline src/EmployerPR/SFA.DAS.EmployerPR.Api/AppStart/AddServiceCollectionExtensions.cs:48]: These two are duplicate, why do we need two anyway?

File: src/Shared/SFA.DAS.SharedOuterApi/InnerApi/Responses/RoatpV2/ProviderDetails.cs [hunk 4]
```diff
@@ -0,0 +1,4 @@
+﻿namespace SFA.DAS.SharedOuterApi.InnerApi.Responses.RoatpV2;
+public class ProviderDetails : Provider
```
Comment (DevenKShah) [inline src/Shared/SFA.DAS.SharedOuterApi/InnerApi/Responses/RoatpV2/ProviderDetails.cs:2]: I dont think this adds any value, we can use Provider class in the Response object.

File: src/EmployerPR/SFA.DAS.EmployerPR/Application/Queries/GetRoatpProviders/GetRoatpProvidersQueryHandler.cs [hunk 5]
```diff
@@ -0,0 +1,21 @@
+﻿using MediatR;
+using SFA.DAS.SharedOuterApi.Interfaces;
+
+namespace SFA.DAS.EmployerPR.Application.Queries.GetRoatpProviders;
+public class GetRoatpProvidersQueryHandler : IRequestHandler<GetRoatpProvidersQuery, GetRoatpProvidersQueryResult>
```
Comment (DevenKShah) [inline src/EmployerPR/SFA.DAS.EmployerPR/Application/Queries/GetRoatpProviders/GetRoatpProvidersQueryHandler.cs:5]: You can now use class level constructors 
`public class GetRoatpProvidersQueryHandler(IRoatpV2TrainingProviderService _roatpV2TrainingProviderService) : IRequestHandler<GetRoatpProvidersQuery, GetRoatpProvidersQueryResult>`

File: src/EmployerPR/SFA.DAS.EmployerPR.Api/Controllers/ProvidersController.cs [hunk 6]
```diff
@@ -0,0 +1,22 @@
+﻿using MediatR;
+using Microsoft.AspNetCore.Mvc;
+using SFA.DAS.EmployerPR.Application.Queries.GetRoatpProviders;
+
+namespace SFA.DAS.EmployerPR.Api.Controllers;
+
+[Route("[controller]")]
+[ApiController]
+public class ProvidersController : ControllerBase
```
Comment (DevenKShah) [inline src/EmployerPR/SFA.DAS.EmployerPR.Api/Controllers/ProvidersController.cs:9]: Use class level constructors

File: src/Shared/SFA.DAS.SharedOuterApi/InnerApi/Responses/RoatpV2/GetProviderSummaryResponse.cs (hunk 1 above)
Comment (MarkFCain) [reply src/Shared/SFA.DAS.SharedOuterApi/InnerApi/Responses/RoatpV2/GetProviderSummaryResponse.cs:3]: it's all deleted, not sure what you mean

File: src/EmployerPR/SFA.DAS.EmployerPR/Domain/RoatpProvider/GetRoatpProvidersResponse.cs (hunk 2 above)
Comment (MarkFCain) [reply src/EmployerPR/SFA.DAS.EmployerPR/Domain/RoatpProvider/GetRoatpProvidersResponse.cs:2]: The incoming json names the collection as RegisteredProviders, so this accommodates that naming conversion

File: src/EmployerPR/SFA.DAS.EmployerPR.Api/AppStart/AddServiceCollectionExtensions.cs (hunk 3 above)
Comment (MarkFCain) [reply src/EmployerPR/SFA.DAS.EmployerPR.Api/AppStart/AddServiceCollectionExtensions.cs:48]: Both are in the shared outer api, and as we discussed, the service wraps the new call.  The service consumes the ApiClient, hence both need to be configured.

File: src/Shared/SFA.DAS.SharedOuterApi/InnerApi/Responses/RoatpV2/ProviderDetails.cs (hunk 4 above)
Comment (MarkFCain) [reply src/Shared/SFA.DAS.SharedOuterApi/InnerApi/Responses/RoatpV2/ProviderDetails.cs:2]: ok

File: src/EmployerPR/SFA.DAS.EmployerPR/Application/Queries/GetRoatpProviders/GetRoatpProvidersQueryHandler.cs (hunk 5 above)
Comment (MarkFCain) [reply src/EmployerPR/SFA.DAS.EmployerPR/Application/Queries/GetRoatpProviders/GetRoatpProvidersQueryHandler.cs:5]: I tried to implement this, and came back with message 'Class cannot have primary constructor'. Visual studio offered no suggestions, and google did not turn up any suggestions.

I think the code is pretty clear as is, so leaving this as is.

Comment (MarkFCain) [reply src/EmployerPR/SFA.DAS.EmployerPR/Application/Queries/GetRoatpProviders/GetRoatpProvidersQueryHandler.cs:5]: I did find some examples and they seem to reflect what you say, but Visual Studio just doesn't work with it, so not changing.

File: src/EmployerPR/SFA.DAS.EmployerPR.Api/Controllers/ProvidersController.cs (hunk 6 above)
Comment (MarkFCain) [reply src/EmployerPR/SFA.DAS.EmployerPR.Api/Controllers/ProvidersController.cs:9]: same issue as the other one, Visual Studio states 'Class cannot have primary constructors' and will not compile. Leaving as is.

**Pull request closed without merging**